python/ray/autoscaler/_private/util.py

```diff
@@ -101,28 +101,29 @@ def prepare_config(config):
 
 def rewrite_legacy_yaml_to_available_node_types(
         config: Dict[str, Any]) -> Dict[str, Any]:
-    if "available_node_types" in config:
-        return config
-    else:
+
+    if "available_node_types" not in config:
         # TODO(ameer/ekl/alex): we can also rewrite here many other fields
         # that include initialization/setup/start commands and ImageId.
+        logger.debug("Converting legacy cluster config to multi node types.")
         config["available_node_types"] = {
             NODE_TYPE_LEGACY_HEAD: {
                 "node_config": config["head_node"],
-                "resources": {},
+                "resources": config["head_node"].get("Resources") or {},
```

Comment:
Why is Resources capitalized here?

Reply:
It seems like this is the use case previously used in tests: https://github.com/ray-project/ray/blob/d1182b827af9fc94be0cd9a6a3211c096aa4c0e3/python/ray/tests/test_autoscaler.py#L614

and also in legacy autoscaler:
https://github.com/ray-project/ray/blob/7b8bd15702d1f0f58794e828c85a7e7a3ec0bc02/python/ray/autoscaler/_private/autoscaler.py#L386

I was also confused with why it is capitalized.

Reply:
Removed this. Capital resources is clearly wrong.